cirq/protocols/commutes.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Protocol for determining commutativity."""
+
+from typing import Any, TypeVar, Union
+
+import numpy as np
+
+from cirq import linalg
+from cirq.type_workarounds import NotImplementedType
+
+TDefault = TypeVar('TDefault')
+
+
+def commutes(left_val: Any,
+             right_val: Any,
+             atol: Union[int, float] = 1e-8,
```

Comment:
Force keywords for `atol` and `default`.

Reply:
Done